client/components/calypso-i18n-provider/index.tsx

```diff
@@ -26,11 +30,11 @@ const CalypsoI18nProvider: React.FunctionComponent = ( { children } ) => {
 		return () => {
 			i18n.off( 'change', onChange );
 		};
-	}, [] );
+	}, [ localeSlug ] );
```

Comment:
We don't need this `localeSlug` dependency. All it does is that it unregisters the `i18n` subscription and adds it again, without changing anything.